### PR TITLE
[ci] Sync template only from current SDK

### PR DESCRIPTION
**Pull request by tchayen**
# Why

We were syncing `main` branch to the external template repo used by expo.new onboarding.

# How

By switching to the current SDK branch.

# Test Plan

If it doesn't work it will not break anything and will be easy to spot and fix.

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [x] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Review by brentvatne — APPROVED** [review]

File: .github/workflows/sync-template.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ name: Sync App Template Repository
 on:
   workflow_dispatch: {}
   push:
-    branches: [main]
+    branches: [sdk-51]
```
Comment (tsapeta) [inline .github/workflows/sync-template.yml:6]: Are we now going to update this as part of each release? Can't we do `sdk-*` here and then make a simple condition to skip steps when the SDK branch is not the latest?

Also, shouldn't we version the template, i.e. have equivalent branches for each SDK?

Comment (brentvatne) [reply .github/workflows/sync-template.yml:6]: > Are we now going to update this as part of each release? Can't we do sdk-* here and then make a simple condition to skip steps when the SDK branch is not the latest?

this is a good idea. we should do this

> Also, shouldn't we version the template, i.e. have equivalent branches for each SDK?

this isn't required -- the only purpose of this template sync is for expo.new, where we always want to use the latest version of the template for the latest sdk release

Comment (tchayen) [reply .github/workflows/sync-template.yml:6]: Good point about `sdk-*`. What's the best way to check if SDK is latest in that context?

---

> the only purpose of this template sync is for expo.new, where we always want to use the latest version of the template for the latest sdk release

It wouldn't hurt though to give users more options in future, similar to how versioning used to work in Expo Go (but I might be missing something and maybe it doesn't make practical sense).

Comment (brentvatne) [reply .github/workflows/sync-template.yml:6]: the canonical way is to check https://exp.host/--/api/v2/versions for the most recent version with a `releaseNoteUrl` 👀 or alternatively check which version has latest tag on npm

Comment (tsapeta) [reply .github/workflows/sync-template.yml:6]: The latter is probably the easiest and more reliable, so something like `npm view expo@latest version --silent | cut -d . -f1` to get the number.

**Pull request merged**